ui/react-components/components/ListView/ListView.jsx

```diff
@@ -0,0 +1,108 @@
+
+                return compareTime(listItem1SortColumn, listItem2SortColumn, sortOrder)
+            } else {
+                const listItem1SortColumn = listItem1[sortColumn] ? listItem1[sortColumn].toLowerCase() : "",
+                    listItem2SortColumn = listItem2[sortColumn] ? listItem2[sortColumn].toLowerCase() : ""
+                return compare(listItem1SortColumn, listItem2SortColumn, sortOrder)
+            }
+        })
+    }
+
+    function compareTime(time1, time2, sortOrder) {
+        if (time1.isBefore(time2)) {
+            return -1 * sortOrder
+        } else if (time1.isAfter(time2)) {
+            return 1 * sortOrder
+        } else {
+            return 0
+        }
+    }
+
+    function compare(item1, item2, sortOrder) {
+        if (item1 < item2) {
+            return -1 * sortOrder
+        } else if (item1 > item2) {
+            return 1 * sortOrder
+        } else {
+            return 0
+        }
+    }
+
+    
+    useEffect(() => {
+        if (ascending) {
+            setListItems(sort(1))
+        } else {
+            setListItems(sort(-1))
+        }
+    }, [ascending, sortColumn])
+
+
+    function renderSortIcon(column) {
+        const caretIcon = ascending ? "fa-caret-up" : "fa-caret-down"
+        if (sortColumn === column) {
+            return (<i data-testid="sortIcon" className={classNames("fa", caretIcon, sortIcon)} onClick={() => setAscending(!ascending)}></i>)
+        }
+        return null
+    }
+
+    if (listItems.length === 0) {
+        return (
+            <div className={classNames(noAppointmentsListView)}>
+                No appointments found
+            </div>)
+    }
+    return (
+        <table className={classNames(appointmentListView)}>
+            <thead>
+                <tr>
+                    {Object.keys(columns).map(column => {
+                        let contentLength = ""
+                        if (columns[column] === "mid") {
+                            contentLength = "table-mid-width"
+                        }
+                        return (
+                            <th key={column} className={classNames(contentLength)} onClick={() => setSortColumn(column)}>
+                                {column}
+                                {renderSortIcon(column)}
+                            </th>
+                        );
+                    })}
+                </tr>
+            </thead>
+            <tbody>
+                {listItems.map((listItem,index) => {
+                    return (<tr key={index}>
+                        {Object.keys(listItem).map((header,index) => {
+
+                            let contentLength = ""
+                            if (columns[header] === "mid") {
+                                contentLength = "table-mid-width"
+                            }
+                            return (<td key={index} className={classNames(contentLength)}>{listItem[header]}</td>)
```

Comment:
We need to highlight the cancelled and missed appointment with different colors. Please create them on product qa to check the behaviour.

Reply:
Made the changes. The component now accepts a handler for colors based on the appointment/listItem